src/app/clusters/thermostat-server/thermostat-server.cpp

```diff
@@ -321,46 +321,21 @@ bool IsPresetHandlePresentInPresets(Delegate * delegate, const ByteSpan & preset
 }
 
 /**
- * @brief Returns the length of the list of presets if the pending presets were to be applied. The calculation is done by
- *        adding the number of presets in Presets attribute list to the number of pending presets in the pending
- *        presets list and subtracting the number of duplicate presets. This is called before changes are actually applied.
+ * @brief Returns the length of the list of presets if the pending presets were to be applied. The size of the pending presets list
+ *        calculated, after all the constraint checks are done, is the new size of the updated Presets attribute since the pending
+ *        preset list is expected to have all existing presets with or without edits plus new presets.
+ *        This is called before changes are actually applied.
  *
  * @param[in] delegate The delegate to use.
  *
  * @return count of the updated Presets attribute if the pending presets were applied to it. Return 0 for error cases.
  */
 uint8_t CountUpdatedPresetsAfterApplyingPendingPresets(Delegate * delegate)
```

Comment:
How about we just rename this to CountNumberOfPendingPresets?  That's what it does, and that's what the caller cares about.

Reply:
done.